lib/full-site-editing/block-templates.php

```diff
@@ -343,7 +343,7 @@ function gutenberg_get_block_template( $id, $template_type = 'wp_template' ) {
 	}
 	list( $theme, $slug ) = $parts;
 	$wp_query_args        = array(
-		'name'           => $slug,
+		'post_name__in'  => array( $slug ),
```

Comment:
using `post_name__in` instead of `name` will allow this query to not be treated as singular, and therefore respect the taxonomy query terms.

Reply:
Interesting :)

Reply:
Can we add a unit test for this? There are already some unit tests for this class.

Reply:
> using post_name__in instead of name will allow this query to not be treated as singular, and therefore respect the taxonomy query terms.

Trying to wrap my head around this. Does it make the query singular because, theoretically, the name is a slug which _should_  be unique to one post? Either way, nice find 👍

Reply:
> Edit: Found the answer in a previous comment. For others that are curious, check out this discussion. In short, yes, the query with name appears to be too specific.

Yup, or this comment https://github.com/WordPress/gutenberg/pull/29842#issuecomment-800667857 points out the same code wise.  Interpereting that leads to the same conclusion, 'name' is seen as so specific that there is no need to check further tax queries.

Reply:
> Can we add a unit test for this? There are already some unit tests for this class.

👍  Updated in deca8c2 - adding another test is not actually necessary, but updating the test setup seems to work.  I created another template post in the setup with the same name but a different theme value.  This gives the current tests a chance to catch these problems (resolving for wrong theme, or unexpected suffix addition on slugs).  Running this updated test class without the changes made to block-templates.php result in multiple tests fail due to finding unexpected theme and slug values as we would expect.

Reply:
> Trying to wrap my head around this. Does it make the query singular because, theoretically, the name is a slug which should be unique to one post?

Yes, I think so. It's the same thing that got me confused [here](https://github.com/WordPress/gutenberg/pull/29842#discussion_r595693853). I think that also explains the existing logic in tax query in core code that Addie pointed out - meaning that it's not a core bug, since there is an expectation of uniqueness.

Which makes me wonder whether we should be allowing duplicate slugs for template parts? (not a blocker for this PR)

Reply:
We could add a comment here explaining _why_ we are using `post_name__in` instead of `name`.